yoti_test.go

```diff
@@ -452,67 +433,9 @@ func TestYotiClient_MissingPostalAddress_UsesFormattedAddress(t *testing.T) {
 	profileAddress, profileErr := ensureAddressProfile(profile)
 	assert.Assert(t, is.Nil(profileErr), "Failed to add formatted address to address on Profile")
 
-	userProfileAddress, userProfileErr := ensureAddressUserProfile(userProfile)
-	assert.Assert(t, is.Nil(userProfileErr), "Failed to add formatted address to address on UserProfile")
-
 	escapedFormattedAddressText := strings.Replace(formattedAddressText, `\n`, "\n", -1)
 
 	assert.Equal(t, profileAddress, escapedFormattedAddressText, "Address does not equal the expected formatted address.")
-	assert.Equal(t, userProfileAddress, escapedFormattedAddressText, "Address does not equal the expected formatted address.")
-
-	var structuredPostalAddress *attribute.JSONAttribute
-	structuredPostalAddress, err = profile.StructuredPostalAddress()
-
-	assert.Assert(t, is.Nil(err))
-	assert.Equal(t, structuredPostalAddress.ContentType, yotiprotoattr.ContentType_JSON, "Retrieved attribute does not have the correct type")
-}
-
-func TestYotiClient_PresentPostalAddress_DoesntUseFormattedAddress(t *testing.T) {
-	var addressText = `PostalAddress`
-
-	var structuredAddressBytes = []byte(`
-	{
-		"address_format": 2,
-		"building": "House No.86-A",
-		"formatted_address": "FormattedAddress"
-	}`)
-	structuredAddress, err := attribute.UnmarshallJSON(structuredAddressBytes)
-
-	assert.Assert(t, is.Nil(err), "Failed to parse structured address")
-
-	var result = UserProfile{
-		ID:                      "remember_me_id0123456789",
-		OtherAttributes:         make(map[string]AttributeValue),
-		StructuredPostalAddress: structuredAddress,
-		Address:                 addressText}
-
-	newFormattedAddress, err := ensureAddressUserProfile(result)
-
-	assert.Assert(t, is.Nil(err), "Failure when getting formatted address")
-	assert.Equal(t, newFormattedAddress, "", "Address should be unchanged when it is present")
-}
-
-func TestYotiClient_MissingFormattedAddress_AddressUnchanged(t *testing.T) {
-	var structuredAddressBytes = []byte(`
-	{
-		"address_format": 2,
-		"building": "House No.86-A"
-	}`)
-
-	structuredAddress, err := attribute.UnmarshallJSON(structuredAddressBytes)
-
-	assert.Assert(t, is.Nil(err), "Failed to parse structured address")
-
-	var result = UserProfile{
-		ID:                      "remember_me_id0123456789",
-		OtherAttributes:         make(map[string]AttributeValue),
-		StructuredPostalAddress: structuredAddress,
-		Address:                 ""}
-
-	address, err := ensureAddressUserProfile(result)
-
-	assert.Assert(t, is.Nil(err), "Failed to add formatted address to address")
-	assert.Equal(t, address, "", "Formatted address missing, but address was still changed")
 }
```

Comment:
Looks like we've removed the only tests which covered lines 247-261 in yoticlient.go (replacing postal address with formatted address, if the former is missing and the latter is present).

Can we add tests for this?

Would be good to get our automated coverage checks added to Go so we can automatically flag things like this too

Reply:
Looks like it was my mistake, those lines weren't covered before. Please would you be able to add a ticket for adding coverage for those lines though (or just adding coverage for that file in general)?